aptos-move/framework/aptos-framework/sources/block.move

```diff
@@ -151,20 +153,66 @@ module aptos_framework::block {
         stake::update_performance_statistics(proposer_index, failed_proposer_indices);
         state_storage::on_new_block(reconfiguration::current_epoch());
 
-        if (timestamp - reconfiguration::last_reconfiguration_time() >= block_metadata_ref.epoch_interval) {
+        block_metadata_ref.epoch_interval
+    }
+
+    /// Set the metadata for the current block.
+    /// The runtime always runs this before executing the transactions in a block.
+    fun block_prologue(
+        vm: signer,
+        hash: address,
+        epoch: u64,
+        round: u64,
+        proposer: address,
+        failed_proposer_indices: vector<u64>,
+        previous_block_votes_bitvec: vector<u8>,
+        timestamp: u64
+    ) acquires BlockResource {
+        let epoch_interval = block_prologue_common(&vm, hash, epoch, round, proposer, failed_proposer_indices, previous_block_votes_bitvec, timestamp);
+        if (timestamp - reconfiguration::last_reconfiguration_time() >= epoch_interval) {
             reconfiguration::reconfigure();
         };
     }
 
+    /// `block_prologue()` but do reconfiguration with DKG after epoch timed out.
+    ///
+    /// Also, it will run as account `0x1` to make on-chain config locking easier.
+    fun block_prologue_ext(
+        account: signer,
+        hash: address,
+        epoch: u64,
+        round: u64,
+        proposer: address,
+        failed_proposer_indices: vector<u64>,
+        previous_block_votes_bitvec: vector<u8>,
+        timestamp: u64,
+        randomness_available: bool,
+        randomness: vector<u8>,
+    ) acquires BlockResource {
+        let epoch_interval = block_prologue_common(&account, hash, epoch, round, proposer, failed_proposer_indices, previous_block_votes_bitvec, timestamp);
+        randomness::on_new_block(&account, randomness_available, randomness);
+
+        if (dkg::in_progress()) {
+            let dkg_timed_out = dkg::update(false, vector[]);
+            if (dkg_timed_out) {
+                // Proceed to the next epoch without randomness.
```

Comment:
Let's discuss if we want to add the following fallback. If DKG timeout and ValidatorSet does not change, we use the old DKG transcript for new epoch.